### PR TITLE
Simplify e2e tests

### PR DESCRIPTION
**Pull request by joshklop**
## Summary by CodeRabbit

- **New Features**
	- Enhanced end-to-end testing capabilities with new command-line options for more granular control over testing configurations.
	- Improved handling of user accounts and deployment configurations for Ethereum-based applications.

- **Bug Fixes**
	- Streamlined user account management to reduce potential errors related to address mismatches in transactions.

- **Refactor**
	- Restructured private key management for clarity and security.
	- Updated configuration management to enhance modularity and reduce dependencies.

- **Chores**
	- Removed obsolete state dump functionality to simplify the codebase.
	- Optimized dependency management by adjusting requirements in the project.

### CODE REVIEW

**Comment by joshklop** [comment]
There are probably even more ways we can simplify the e2e tests, but this was the low-hanging fruit.

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,11 @@ test-all:
 
 .PHONY: e2e
 e2e:
-	go test -v ./e2e
+	go test -v ./e2e \
+	-l1-allocs ./optimism/.devnet/allocs-l1.json \
+	-l2-allocs-dir ./optimism/.devnet/ \
+	-l1-deployments ./optimism/.devnet/addresses.json \
+	-deploy-config ./optimism/packages/contracts-bedrock/deploy-config/devnetL1.json
```
[the remark this answers is not in the document]
Comment (joshklop) [reply Makefile:26]: That's an interesting idea. Maybe we should just copy these files to the Monomer repo and remove the submodule entirely. Wdyt?

[the remark this answers is not in the document]
Comment (natebeauregard) [reply Makefile:26]: Yeah I like that a lot, if we're able to remove the submodule then our repo would definitely be more digestible. I feel like that update can probably be done in a separate PR unless you feel inspired to include it in this one

[the remark this answers is not in the document]
Comment (joshklop) [reply Makefile:26]: https://github.com/polymerdao/monomer/issues/222

File: e2e/op_stack.go [hunk 2]
```diff
@@ -85,35 +85,23 @@ func (op *OPStack) Run(ctx context.Context, env *environment.Env) error {
 	}
 
 	// check balance of privKey - operator must hold an L1 balance to post state roots & batches
```
[the remark this answers is not in the document]
Comment (joshklop) [reply e2e/op_stack.go:87]: I'll just remove this entire balance check.

**Pull request merged**